vaadin-crud-flow-parent/vaadin-crud-flow/src/main/java/com/vaadin/flow/component/crud/Crud.java

```diff
@@ -190,25 +190,20 @@ public Crud() {
         deleteButton.getElement().setAttribute("slot", "delete-button");
         deleteButton.addThemeNames("tertiary", "error");
         getElement().appendChild(deleteButton.getElement());
-
-        addAttachListener(e -> {
-            getElement()
-                    .executeJs("this.__validate = function () {return true;}");
-            // Override onFormChanges to dispatch an event so the server can
-            // control the dirty state of the form
-            getElement().executeJs(
-                    "this.__onFormChanges = function () { this.dispatchEvent(new CustomEvent('form-change')); }");
-        });
-
-        getElement().addEventListener("form-change", this::formChangeEvent);
     }
 
-    private void formChangeEvent(com.vaadin.flow.dom.DomEvent e) {
-        if (this.saveButton.isEnabled()) {
-            this.setDirty(true);
+    private class SaveButton extends Button {
+        @Override
+        public void onEnabledStateChanged(boolean enabled) {
+            super.onEnabledStateChanged(enabled);
+            overrideSaveDisabled(enabled);
         }
     }
 
+    private void overrideSaveDisabled(boolean enabled) {
```

Comment:
Consider moving this into the SaveButton class

Reply:
I have considered it initially, but it's easier to override the `__isSaveBtnDisabled` from the Crud scope without introducing new fields/getters 🤔